crates/hotshot/src/traits/election/static_committee.rs

```diff
@@ -95,24 +104,40 @@ where
         }
     }
 
-    fn default_election_config(num_nodes: u64) -> TYPES::ElectionConfigType {
-        StaticElectionConfig { num_nodes }
+    fn default_election_config(
+        num_nodes_with_stake: u64,
+        num_nodes_without_stake: u64,
+    ) -> TYPES::ElectionConfigType {
+        StaticElectionConfig {
+            num_nodes_with_stake,
+            num_nodes_without_stake,
+        }
     }
 
     fn create_election(
         entries: Vec<PeerConfig<PUBKEY>>,
         config: TYPES::ElectionConfigType,
     ) -> Self {
+        // get nodes with stake from `known_nodes_with_stake`
         let nodes_with_stake: Vec<PUBKEY::StakeTableEntry> = entries
             .iter()
             .map(|x| x.stake_table_entry.clone())
             .collect();
+        // make committee_nodes_with_stake to be nodes_with_stake
         let mut committee_nodes_with_stake: Vec<PUBKEY::StakeTableEntry> = nodes_with_stake.clone();
         debug!("Election Membership Size: {}", config.num_nodes);
-        committee_nodes_with_stake.truncate(config.num_nodes.try_into().unwrap());
+        // truncate committee_nodes_with_stake to only `num_nodes`
+        // since the `num_nodes_without_stake` are not part of the committee,
+        committee_nodes_with_stake.truncate(config.num_nodes_with_stake.try_into().unwrap());
+
+        // get the non-staked pub keys from the config's `known_nodes_without_stake`
+        // it is not option anymore
+        committee_nodes_without_stake = config.known_nodes_without_stake.clone();
```

Comment:
@bfish713 How do we go about it? I think this is wrong. config is ElectionConfigType, which has num_nodes_with_stake/without stake, so how to get the `committee_nodes_without_stake` from it, which is Vec<PUBKEY>, can we do a similar call i.e `x.get_stake_table_entry()` as above?

Reply:
It is because we don't to change the function signatures

Reply:
I think you want to filter the entries by stake to split the list into those with/without stake.  Entries should have both entries with 1 and 0 stake.

Reply:
I have tried, let me know now